libs/SalesforceSDKCore/SalesforceSDKCore/Classes/RestAPI/SFRestAPI.m

```diff
@@ -197,6 +186,7 @@ - (void)send:(SFRestRequest *)request delegate:(id<SFRestDelegate>)delegate shou
         [[SFAuthenticationManager sharedManager] loginWithCompletion:^(SFOAuthInfo *authInfo, SFUserAccount *userAccount) {
             __strong typeof(weakSelf) strongSelf = weakSelf;
             [strongSelf enqueueRequest:request delegate:delegate shouldRetry:shouldRetry];
+            [SFUserAccountManager sharedInstance].currentUser = userAccount;
```

Comment:
NB: what do you guys think about making this authentication block go away in the future (6.0?), in favor of allowing unauthenticated requests to fail in this scenario?  In our quest to minimize the setting of `currentUser`, this "convenience authentication" block gives me pause.  It means that the user has bypassed the SDK manager launch process, and IMO that's something that should be discouraged.  Thoughts?